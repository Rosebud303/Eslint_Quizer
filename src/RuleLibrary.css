.rules {
    display: flex;
    justify-content: space-around;
    align-content: center;
    height: 400px;
}

.rule:hover {
    cursor: pointer;
    background: #78b6ec;
}

.rule {
    width: 225px;
    height: 100px;
    display: -webkit-flex;
    display: flex;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-align-items: center;
    align-items: center;
    border: 2px solid;
    border-radius: 25px;
    box-shadow: 4px 4px 2px #1d1b1b;
    background: #68e277;
}