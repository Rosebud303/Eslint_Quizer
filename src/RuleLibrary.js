import React, { Component } from 'react';
import CategoryDescription from './CategoryDescription.js';
import './RuleLibrary.css'

export default class RuleLibrary extends Component {
  constructor(props) {
      super(props);
      this.state = {
          isDisplayed: false,
          selectedCategory: {},
          currentCategory: ''
      }  
  }

  getCurrent = (e) => {
      this.setState({currentCategory: e.target.innerText}, () => {
          this.findSelected();
      });
      this.toggleDisplay();
  }

  findSelected = () => {
      let newSelect = this.props.categories.find( category => {
          return category.name === this.state.currentCategory
      });
      this.setState({selectedCategory: newSelect});
  };

  toggleDisplay = () => {
      this.setState({isDisplayed: !this.state.isDisplayed});
  };
    
  render() {
    return (
      <div className='rules'>
        {
          this.props.categories.map(category => {
              return (
                <h2 key={category.name} 
                    className='rule' 
                    onClick={this.getCurrent}>
                    {category.name}
                </h2>
              );
          })
        }
        { this.state.isDisplayed && 
            <CategoryDescription selectedCat={this.state.selectedCategory}
                                 toggle={this.toggleDisplay}
            />
        }
      </div>
    )
  };
};
