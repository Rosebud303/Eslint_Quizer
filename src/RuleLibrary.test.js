import React from 'react';
import RuleLibrary from './RuleLibrary.js';
import { shallow } from 'enzyme';

const mockCata =[
    {
        "name": "Indentation",
        "description": "When refering to 'Indentation' in esLint this means we are talking about the stylistic way your code should look in the developer. Indentation makes your code more readable and is the first step into writing less spaghetti code. It also gives the reader visual ques for hierarchy and scoping.",
        "incorrectImg": "https://i.imgur.com/oR17Hke.png",
        "correctImg": "https://i.imgur.com/YIp5OaH.png",
        "url": "https://eslint.org/docs/rules/indent"
    },
    {
        "name": "Key Spacing",
        "description": "When refering to 'Keyword/Function Spacing' in esLint we are talking about the structural style of a line of code. Spacing much like Indentation helps you create more readable code. Spacing after declaration of a key word or function allows the compiler to understand what kind of code you are writing. It can also determine if a piece of code you wrote works as well.",
        "incorrectImg": "https://i.imgur.com/2sbdf12.png",
        "correctImg": "https://i.imgur.com/VX72PJh.png",
        "url": "https://eslint.org/docs/rules/key-spacing"
    }
];

describe('RuleLibrary', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <RuleLibrary categories={mockCata}/>
        )
    });

    it('Should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should have default state', () => {
        expect(wrapper.state()).toEqual({isDisplayed: false, selectedCategory: {}, currentCategory: '' })
    });

    it('Should set state of selectedCategory depending on category name', () => {
        wrapper.setState({currentCategory: 'Indentation'});
        expect(wrapper.state('selectedCategory')).toEqual({});
        wrapper.instance().findSelected();
        expect(wrapper.state('selectedCategory')).toEqual(    {
            "name": "Indentation",
            "description": "When refering to 'Indentation' in esLint this means we are talking about the stylistic way your code should look in the developer. Indentation makes your code more readable and is the first step into writing less spaghetti code. It also gives the reader visual ques for hierarchy and scoping.",
            "incorrectImg": "https://i.imgur.com/oR17Hke.png",
            "correctImg": "https://i.imgur.com/YIp5OaH.png",
            "url": "https://eslint.org/docs/rules/indent"
        });
    });

    it('Should toggle the state of isDisplayed', () => {
        expect(wrapper.state('isDisplayed')).toEqual(false);
        wrapper.instance().toggleDisplay();
        expect(wrapper.state('isDisplayed')).toEqual(true);
    });

    it('Should set state of currentCategory with event.target.InnerText', () => {
        const mockTarget = {target: {innerText: 'e'}}
        expect(wrapper.state('currentCategory')).toEqual('');
        wrapper.instance().getCurrent(mockTarget);
        expect(wrapper.state('currentCategory')).toEqual('e')
    });
});