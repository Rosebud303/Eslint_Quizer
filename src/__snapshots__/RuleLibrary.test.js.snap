// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleLibrary Should match the snapshot 1`] = `
<div
  className="rules"
>
  <h2
    className="rule"
    key="Indentation"
    onClick={[Function]}
  >
    Indentation
  </h2>
  <h2
    className="rule"
    key="Key Spacing"
    onClick={[Function]}
  >
    Key Spacing
  </h2>
</div>
`;
